lib/trino-filesystem/src/main/java/io/trino/filesystem/Write.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Represents ongoing write operation.
+ */
+public interface Write
+        extends AutoCloseable
+{
+    /**
+     * The output stream to write to. The caller is not responsible for closing the stream.
+     *
+     * @see #close()
+     */
```

Comment:
Is the caller responsible for closing the `Write` object? Which in turn is responsible for closing the stream returned by this method?